### PR TITLE
Update release.md guide

### LINKED ISSUE #4347: Update release guide
Update release guide to fix some errors and add a troubleshooting section for github actions

### PR DESCRIPTION
**Pull request by MichaelFu512**
### Pull Request Description
Closes #4347

Added section on troubleshooting github action
Fixed some formatting and words

-----
*After creating the pull request: in order to pass the **release_notes_updated** check you will need to update the "Future Release" section of* `docs/source/release_notes.rst` *to include this pull request by adding :pr:`123`.*

### CODE REVIEW

**Comment by machineFL** [comment]
No linked issues found. Please add the corresponding issues in the pull request description. <br/> [Use GitHub automation to close the issue when a PR is merged](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

File: docs/source/release_notes.rst [hunk 1]
```diff
@@ -6,6 +6,7 @@ Release Notes
     * Changes
         * Pinned networkx version below 3.2 for Python version compatibility :pr:`4351`
     * Documentation Changes
+      * Update release guide :pr:`4346`
```
Comment (eccabay) [inline docs/source/release_notes.rst:9]: Nitpick: can this be past tense and slightly more detailed?

File: release.md [hunk 2]
```diff
@@ -121,3 +126,16 @@ Per the instructions [here](https://conda-forge.org/docs/maintainer/updating_pkg
     - Select *Bot commands* and put the following title (change `username`): `@conda-forge-admin, please add user @username`
 2. A PR will be auto-created on the repo, and will need to be merged by an existing maintainer.
 3. The new user will need to **check their email for an invite link to click**, which should be https://github.com/conda-forge
+
+## Troubleshooting failed Github actions
+
+If some Github actions are failing and they aren't automatically ran in the PR request, to easily test them without having to push to main:
```
Comment (eccabay) [inline release.md:132]: Mega grammar nit: `and they aren't automatically run in the PR request` -> `and they aren't automatically run in the PR's automated tests`

**Review by eccabay — APPROVED** [review]
Just some non-blocking nitpicks, thanks!

**Review by jeremyliweishih — APPROVED** [review]

**Pull request merged**